prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiverFilter.kt

```diff
@@ -28,24 +57,397 @@ class FHIRReceiverFilter(
+
+                azureEventService.trackEvent(
+                    ReportRouteEvent(
+                        queueMessage.reportId,
+                        report.id,
+                        queueMessage.topic,
+                        sender,
+                        receiver.fullName,
+                        receiverObservationSummary,
+                        filteredObservationSummary,
+                        bodyString.length,
+                        AzureEventUtils.getIdentifier(receiverBundle)
+                    )
+                )
+
+                listOf(
+                    FHIREngineRunResult(
+                        nextEvent,
+                        report,
+                        blobInfo.blobUrl,
+                        FhirTranslateQueueMessage(
+                            report.id,
+                            blobInfo.blobUrl,
+                            BlobAccess.digestToString(blobInfo.digest),
+                            queueMessage.blobSubFolderName,
+                            queueMessage.topic,
+                            receiver.fullName
+                        )
+                    )
+                )
+            } else {
+                logger.info("Report did not pass receiver filters. Terminating lineage.")
+                // this bundle does not have receivers; only perform the work necessary to track the routing action
+                // create none event
+                val nextEvent = ProcessEvent(
+                    Event.EventAction.NONE,
+                    queueMessage.reportId,
+                    Options.None,
+                    emptyMap(),
+                    emptyList()
+                )
+                val emptyReport = Report(
+                    Report.Format.FHIR,
+                    emptyList(),
+                    1,
+                    metadata = this.metadata,
+                    topic = queueMessage.topic
+                )
+
+                // create item lineage
+                emptyReport.itemLineages = listOf(
+                    ItemLineage(
+                        null,
+                        queueMessage.reportId,
+                        1,
+                        emptyReport.id,
+                        1,
+                        null,
+                        null,
+                        null,
+                        emptyReport.getItemHashForRow(1)
+                    )
+                )
+
+                // ensure tracking is set
+                actionHistory.trackCreatedReport(nextEvent, emptyReport)
+
+                val observationSummary = AzureEventUtils.getObservationSummaries(bundle)
+                azureEventService.trackEvent(
+                    ReportRouteEvent(
+                        queueMessage.reportId,
+                        emptyReport.id,
+                        queueMessage.topic,
+                        sender,
+                        null,
+                        observationSummary,
```

Comment:
Having observationSummary here for both is not necessarily correct, right? Maybe no observations were filtered but some other filter failed? Maybe we want a different event here, like "ReportFilteredOutEvent" that includes the filtered error message and correctly indicated what observations were actually filtered out?

An alternative to a new event is to include "ErrorMsg" as a param here and just leave it null when a report is not filtered out.